sdk/scheduler/src/main/java/com/mesosphere/sdk/api/EndpointUtils.java

```diff
@@ -0,0 +1,150 @@
+     */
+    public static String toVipEndpoint(String serviceName, VipInfo vipInfo) {
+        String hostname = String.format("%s.%s.%s",
+                removeSlashes(vipInfo.getVipName()), removeSlashes(serviceName), VIP_HOST_TLD);
+        return toEndpoint(hostname, vipInfo.getVipPort());
+    }
+
+    /**
+     * Returns the correct L4LB VIP hostname for accessing the Scheduler API given the provided service name.
+     */
+    public static String toSchedulerApiVipHostname(String serviceName) {
+        return String.format("api.%s.marathon.%s", removeSlashes(serviceName), VIP_HOST_TLD);
+    }
+
+    /**
+     * Returns a {@link Label} which can be included in the {@code org.apache.mesos.Protos.DiscoveryInfo} for a VIP.
+     * This is the inverse of {@link #parseVipLabel(String, Label)}
+     */
+    public static Label createVipLabel(String vipName, int vipPort) {
+        return Label.newBuilder()
+                .setKey(String.format("%s%s", VIP_LABEL_PREFIX, UUID.randomUUID().toString()))
+                .setValue(String.format("%s:%d", vipName, vipPort))
+                .build();
+    }
+
+    /**
+     * Extracts VIP information from the provided VIP label.
+     * This is the inverse of {@link #createVipLabel(String, int)}.
+     *
+     * @param taskName task name for use in logs if there's a problem
+     * @param label a label from a {@code org.apache.mesos.Protos.DiscoveryInfo}
+     * @return the VIP information or an empty Optional if the provided label is invalid or inapplicable
+     */
+    public static Optional<VipInfo> parseVipLabel(String taskName, Label label) {
+        if (!label.getKey().startsWith(VIP_LABEL_PREFIX)) {
+            return Optional.empty();
+        }
+
+        // Expected VIP label format: "<vipname>:<port>"
+        List<String> namePort = Splitter.on(':').splitToList(label.getValue());
+        if (namePort.size() != 2) {
+            LOGGER.error("Task {}'s VIP value for {} is invalid, expected 2 components but got {}: {}",
+                    taskName, label.getKey(), namePort.size(), label.getValue());
+            return Optional.empty();
+        }
+        int vipPort;
+        try {
+            vipPort = Integer.parseInt(namePort.get(1));
+        } catch (NumberFormatException e) {
+            LOGGER.error(String.format(
+                    "Unable to Task %s's VIP port from %s as an int",
+                    taskName, label.getValue()), e);
+            return Optional.empty();
+        }
+        return Optional.of(new VipInfo(namePort.get(0), vipPort));
+    }
+
+    /**
+     * "/group1/group2/group3/group4/group5/kafka" => "group1group2group3group4group5kafka".
+     */
+    private static String removeSlashes(String serviceName) {
+        return serviceName.replace("/", "");
+    }
+
+    /**
+     * "/group1/group2/group3/group4/group5/kafka" => "kafka-group5-group4-group3-group2-group1.marathon.mesos".
```

Comment:
This is effectively reproducing what navstar/mesos-dns currently produce for a given service name. The mesos-dns hostnames will likely go away via #1047